File: packages/jetbrains-plugin/src/main/kotlin/com/evalops/composer/toolwindow/RuntimeStatus.kt
package com.evalops.composer.toolwindow

import com.evalops.composer.api.AgentEvent

fun formatRuntimeStatus(event: AgentEvent): String? {
    return when (event) {
        is AgentEvent.Status -> {
            val status = event.status.trim()
            if (status.isEmpty()) {
                null
            } else if (status == "compacting") {
                "Compacting conversation..."
            } else {
                "Status: $status"
            }
        }
        is AgentEvent.Compaction -> {
            if (event.auto == true) {
                "Compacted conversation automatically"
            } else {
                "Compacted conversation"
            }
        }
        else -> null
    }
}

File: packages/web/src/services/runtime-status.ts
import type { AgentEvent } from "./api-client.js";

type RuntimeStatusEvent = Extract<
	AgentEvent,
	{ type: "status" | "compaction" | "tool_batch_summary" }
>;

export function formatWebRuntimeStatus(
	event: RuntimeStatusEvent,
): string | null {
	switch (event.type) {
		case "status": {
			const status = event.status.trim();
			if (!status) {
				return null;
			}
			if (event.details.kind === "tool_execution_summary") {
				return status;
			}
			return status === "compacting"
				? "Compacting conversation..."
				: `Status: ${status}`;
		}
		case "compaction":
			return event.auto
				? "Compacted conversation automatically"
				: "Compacted conversation";
		case "tool_batch_summary":
			return event.summary.trim() || null;
	}
}
